src/tm-threads.c

```diff
@@ -366,6 +366,93 @@ static void *TmThreadsSlotPktAcqLoop(void *td)
     return NULL;
 }
 
+static void *TmThreadsLib(void *td)
+{
+    ThreadVars *tv = (ThreadVars *)td;
+    TmSlot *s = tv->tm_slots;
+    TmEcode r = TM_ECODE_OK;
+    TmSlot *slot = NULL;
+
+    /* Set the thread name */
+    SCSetThreadName(tv->name);
+
+    if (tv->thread_setup_flags != 0)
+        TmThreadSetupOptions(tv);
+
+    /* Drop the capabilities for this thread */
+    SCDropCaps(tv);
+
```

Comment:
@amirabell Is there a reason ` CaptureStatsSetup(tv);` isn't done here like it is done in `TmThreadsSlotPktAcqLoop`, I'm trying refactor the common bits of the existing a new function.

Reply:
Hey @jasonish no reason at all. I actually missed that function call as it was introduced in recent versions of suricata.
Feel free to add it, thanks!